doc/winrunner.rst

```diff
@@ -1,81 +1,460 @@
+reconfigure this environment variable::
+
+    export RACKER_WDM_PROVIDER=vmware_workstation
+
+Possible values are, in alphabetical order, ``hyperv``, ``virtualbox``,
+``qemu``, ``vmware_fusion``, ``vmware_workstation``.
+
+Please note that this has not been tested with providers other than
+`VirtualBox`_, so we would welcome to receive feedback from the community
+whether this also works well for them on other hypervisors.
 
-*****
-Usage
-*****
 
-Which shell spawns faster?
+VM machine
+==========
+
+The architecture of Windows leads to container compatibility requirements that
+are different than on Linux, more background about this detail can be found at
+`Windows container version compatibility`_.
+
+In order to provide appropriate convenience, Racker's launcher subsystem
+inquires the ``os.version`` attribute of the OCI image about the designated
+version of Windows version before starting the container. Based on the version,
+the corresponding Windows Docker Machine host is selected to run the payload
+on. Currently, the supported operating systems are Windows 2016, 2019, and 2022.
+
+Certain container images can still be launched on mismatching operating system
+versions, for example, the `eclipse-temurin`_ container images. By default,
+when possible, the image will be launched on a Windows 2022 machine. If you
+want to explicitly control on which runner host the container will be launched,
+use another environment variable::
+
+    export RACKER_WDM_MACHINE=2019-box
+
+If you receive error messages like ``docker: no matching manifest for
+windows/amd64 10.0.17763 in the manifest list entries.``, reset this setting
+by typing::
+
+    unset RACKER_WDM_MACHINE
+
+
+********
+Examples
+********
+
+
+Introduction
+============
+
+For understanding some of the acronyms used in the following section, it is
+good to memorize those:
+
+- LTSC: Long-Term Servicing Channel
+- SAC: Semi-Annual Channel
+
+For more details, see `Overview of System Center release options`_.
+
+
+Choosing a base image
+=====================
+
+Quoting from `Windows Container Base Images`_:
+
+    How do you choose the right base image to build upon? For most users, Windows
+    Server Core and Nanoserver will be the most appropriate image to use. Each
+    base image is briefly described below:
+
+    - ``Nano Server`` is an ultralight Windows offering for new application
+      development.
+    - ``Server Core`` is medium in size and a good option for "lifting and
+      shifting" Windows Server apps.
+    - ``Windows Server`` has full Windows API support, and allows you to use
+      more server features.
+    - ``Windows`` is the largest image and has full Windows API support for
+      workloads.
+
+The examples outlined within this section will use different Windows container
+images. According to the feature set outlined above, their download sizes are
+different.
+
+- Nano Server: 125 MB
+- Server Core: 2.2 GB
+- Windows Server: 4.8 GB
+- Windows: 7.1 GB
```

Comment:
Good to learn about the container image sizes. But what about the sizes of the Vagrant box image?

Reply:
Added to f49c209658ca2.